R/ObjectiveTuning.R

```diff
@@ -93,6 +85,22 @@ ObjectiveTuning = R6Class("ObjectiveTuning",
       } else {
         aggr[, y, with = FALSE]
       }
+    },
+
+    .resampling = NULL
+  ),
+
+  active = list(
+
+    #' @field resampling ([mlr3::Resampling]).
+    resampling = function(rhs) {
```

Comment:
Is this change necessary for irace or just a general "improvement"?

Reply:
Is it documented that the resampling is instantiated once the objective is created? This is kind of a havy change as now the function is not as "random" anymore.

Reply:
If we leave `resample` a field, a tuner could change it to an uninstantiated resampling and each objective function call would be evaluated on different train/ test splits. Currently, we instantiate the resampling during the initialization. It is a general improvement since we have to pass already instantiated resamplings to irace (see comment bellow). We could discus a different way to do this in #169.